typescript/infra/src/utils/violation.ts

```diff
@@ -0,0 +1,258 @@
+      sortedObj[key] = sortArraysByType(obj[key]);
+    });
+    return sortedObj;
+  }
+  return obj;
+}
+
+function removeFields(obj: any): any {
+  if (Array.isArray(obj)) {
+    return obj.map(removeFields);
+  } else if (obj !== null && typeof obj === 'object') {
+    return Object.keys(obj).reduce((result: any, key: string) => {
+      if (!ignoreFields.includes(key)) {
+        result[key] = removeFields(obj[key]);
+      }
+      return result;
+    }, {});
+  } else {
+    return obj;
+  }
+}
+
+function logDiff(expected: AnyObject, actual: AnyObject): void {
+  const sortedExpected = sortArraysByType(expected);
+  const sortedActual = sortArraysByType(actual);
+
+  const sortedExpectedJson = stringify(sortedExpected, { space: 2 });
+  const sortedActualJson = stringify(sortedActual, { space: 2 });
+
+  const parsedSortedExpected = JSON.parse(sortedExpectedJson);
+  const parsedSortedActual = JSON.parse(sortedActualJson);
+
+  const added = addedDiff(parsedSortedActual, parsedSortedExpected);
+  const deleted = deletedDiff(parsedSortedActual, parsedSortedExpected);
+  const updated = updatedDiff(parsedSortedActual, parsedSortedExpected);
+
+  const logChanges = (
+    changes: AnyObject,
+    changeType: ChangeType,
+    color: (text: string) => string,
+    config: AnyObject,
+    otherConfig?: AnyObject,
+  ) => {
+    const logChange = (obj: AnyObject, path: string = '') => {
+      Object.keys(obj).forEach((key) => {
+        const currentPath = path ? `${path}.${key}` : key;
+        if (
+          typeof obj[key] === 'object' &&
+          obj[key] !== null &&
+          !Array.isArray(obj[key])
+        ) {
+          logChange(obj[key], currentPath);
+        } else {
+          if (changeType !== ChangeType.Updated) {
+            console.log(
+              color(
+                `${changeTypeMapping[changeType]} ${updatePath(
+                  config,
+                  currentPath,
+                )}: ${stringify(obj[key], {
+                  space: 2,
+                })}`,
+              ),
+            );
+          } else {
+            console.log(
+              color(
+                `${changeTypeMapping[changeType]} ${updatePath(
+                  config,
+                  currentPath,
+                )}: ${stringify(getElement(otherConfig, currentPath), {
+                  space: 2,
+                })} -> ${stringify(obj[key], { space: 2 })}`,
+              ),
+            );
+          }
+        }
+      });
+    };
+    logChange(changes);
+  };
+
+  logChanges(added, ChangeType.Added, chalk.green, parsedSortedActual);
+  logChanges(deleted, ChangeType.Deleted, chalk.red, parsedSortedActual);
+  logChanges(
+    updated,
+    ChangeType.Updated,
+    chalk.yellow,
+    parsedSortedExpected,
+    parsedSortedActual,
+  );
+}
+
+function preProcessConfig(config: any) {
+  return removeFields(toLowerCaseValues(config));
+}
+
+function logViolationDetail(violation: CheckerViolation): void {
```

Comment:
nit: maybe nice to put in the fn name or a comment that this only works for ISMs, otherwise you just get that from the use of `IsmConfigSchema`